layrry-launcher/src/main/java/org/moditect/layrry/Layrry.java

```diff
@@ -15,44 +15,29 @@
  */
 package org.moditect.layrry;
 
-import java.io.File;
-
 import org.moditect.layrry.config.LayersConfig;
 import org.moditect.layrry.config.LayersConfigLoader;
-import org.moditect.layrry.internal.Args;
 import org.moditect.layrry.internal.LayersFactory;
 
-import com.beust.jcommander.JCommander;
+import java.io.File;
 
 /**
- * The main entry point for using Layrry. Expects the layers config file to be passed in:
+ * The main entry point for using Layrry in embedded mode. Expects the layers config file to be passed in:
  * <p>
  * <code>
- * Layrry --layers-config &lt;path/to/layrry.yml&gt;
+ * Layrry.run(&lt;path/to/layrry.yml&gt;, &lt;args&gt;)
  * </code>
  */
-public class Layrry {
-
-    public static void main(String... args) throws Exception {
-        Args arguments= new Args();
-
-        JCommander.newBuilder()
-            .addObject(arguments)
-            .build()
-            .parse(args);
-
-        File layersConfigFile = arguments.getLayersConfig().getAbsoluteFile();
+public final class Layrry {
 
+    public static void run(File layersConfigFile, String... args) {
```

Comment:
Should use `Path`.

That said, I'm a bit confused :) If we still have the reference to config parsers in this module, what do we actually gain by this split?

Reply:
The `layrry-core` module defines the core API but does not depend on an specific config format. This lets developers embed Layrry in their custom launchers and pick/force a config format if they want to.

The `layrry-launcher` provides a ready made launcher with predefined config formats, in a single fat JAR.

There are 2 ways to launch a Layrry powered app:
1. provided a configuration file that bootstraps the `Layers` instance to use.
2. provide a programmatically configured `Layers` instance.

The `Layrry` class takes care of 1) while `Layers` takes care of 2) as the `Layers` class already exposes a method for running the computed layers. But, if `Layrry` is to become the single entry point then it should have a method such as

    void run(Layers layers, String main, String... args)